File: solutions/cpp/pangram/2/pangram.cpp
#include "pangram.h"
#include <array>
#include <cctype>

namespace pangram {

bool is_pangram(const std::string& input)
{
    std::array<bool, 26> pangram_map = { 0 };
    for (auto it: input)
    {
        if (std::isupper(it))
        {
            pangram_map[it - 'A'] = true;
        }
        if (std::islower(it))
        {
            pangram_map[it - 'a'] = true;
        }
    }
    for (auto it: pangram_map)
    {
        if (!it)
        {
            return false;
        }
    }
    return true;
}

}  // namespace pangram

File: solutions/go/pangram/2/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
	m := make(map[rune]bool)
    for _, v := range strings.ToLower(input) {
        if 'a' <= v && v <= 'z' {
            m[v] = true
        }
    }
    var ok bool
    for r := 'a'; r <= 'z'; r++ {
        if ok = m[r]; !ok {
            return ok
        }
    }
    return ok
}
